src/include/mpir_request.h

```diff
@@ -226,6 +226,9 @@ struct MPIR_Request {
             MPL_atomic_int_t active_flag;       /* flag indicating whether in a start-complete active period.
                                                  * Value is 0 or 1. */
         } part;                 /* kind : MPIR_REQUEST_KIND__PART_SEND or MPIR_REQUEST_KIND__PART_RECV */
+        struct {
+            MPIR_Win *win;
+        } rma;                  /* kind : MPIR_REQUEST_KIND__RMA */
```

Comment:
I think I went this direction because the `comm` pointer is so ubiquitous and I didn't really want to disrupt existing code using it. If we end up putting more RMA-related info in this top layer, we can reconsider. But I prefer this low-impact change for now.

Maybe we'll be using anonymous union by the time we need it 😄.

Reply:
I think that's fair. Changing the communicator would be a huge impact.